### PR TITLE
net.sh: Add support for selecting validator GPU mode

### LINKED ISSUE #6100: net.sh always runs the CUDA validator implementation if the host has a GPU installed
#### Problem

`net.sh` always runs the CUDA validator implementation if the host has a GPU installed. On the colo nodes, this is always.

#### Proposed Solution

Extend the following logic to respect the `-g/-G` flags to `colo.sh`
https://github.com/solana-labs/solana/blob/093b5b52674f8ef5cb9d725ff3d95ff249c5d59d/net/remote/remote-node.sh#L116-L119

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

There is currently no way to switch between launching a CPU and GPU testnet without reallocating the underlying instances.  If the instances have GPUs, a CUDA-enabled validators are launched, if not CPU-only validators.  This is tedious with GCE and untenable on colo where most of the nodes just *have* GPUs.

#### Summary of Changes

Adds a flag to `net/net.sh` to allow selecting the GPU mode.  Defaults to "auto", which is the current behavior, as described above.

WIP: Until I can get a hold of a couple GPU nodes on colo for testing

Fixes #6100

### CODE REVIEW

**Comment by danpaul000** [comment]
Looks good.  You should be able to test this on GCE if colo nodes not available.  The workflow should be identical, no?

**Comment by t-nelson** [comment]
Mostly, yes.  None of the "startup-script" stuff is run on colo, 'cause we can set it all up at OS install time.  I just want to be sure there's no unexpected interaction there

**Comment by t-nelson** [comment]
Confirmed this works as expected on both GCE and Colo

File: net/net.sh [hunk 1]
```diff
@@ -220,6 +230,9 @@ while getopts "h?T:t:o:f:rD:c:Fn:i:d" opt "${shortArgs[@]}"; do
   n)
     numFullnodesRequested=$OPTARG
     ;;
+  g)
```
Comment (mvines) [inline net/net.sh]: IMO just add `--gpu-mode`.  Short opts are pretty unreadable

File: net/net.sh [hunk 2]
```diff
@@ -284,6 +297,8 @@ while getopts "h?T:t:o:f:rD:c:Fn:i:d" opt "${shortArgs[@]}"; do
   esac
 done
 
+grep -qE '^(on|off|auto|cuda)$' <<<"$gpuMode" || ( echo "Unexpected GPU mode: \"$gpuMode\"" && exit 1 )
```
Comment (mvines) [inline net/net.sh]: with `-g` gone, this check can move into the `--gpu-mode` if block above.  I'd prefer you use pure bash instead of forking off a grep

**Pull request merged**